ftest/src/test/java/org/arquillian/reporter/GreeterReportVerifier.java

```diff
@@ -0,0 +1,229 @@
+        if (subReport.getName().equals(subReportName)) {
+            return true;
+        }
+        return false;
+    }
+
+    private boolean isContainerConfigurationSubReport(Report subReport) {
+        StringKey subReportName = new UnknownStringKey("containers");
+        if (subReport.getName().equals(subReportName)) {
+            return true;
+        }
+        return false;
+    }
+
+    @Test
+    public void verify_test_class_report_exists_with_contents() {
+        List<TestClassReport> testClassReports = executionReport.getTestSuiteReports().get(0).getTestClassReports();
+        TestClassReport testClassReport = testClassReports.get(0);
+        ConfigurationReport configurationReport = testClassReport.getConfiguration();
+        List<TestMethodReport> testMethodReports = testClassReport.getTestMethodReports();
+
+        assertThat(testClassReports).size().isEqualTo(1);
+
+        assertThatTestClassReport(testClassReport)
+            .hasName("org.arquillian.reporter.GreeterTest")
+            .hasNumberOfEntries(1)
+            .hasEntryContainingKeys(ArquillianCoreKey.TEST_CLASS_REPORT_MESSAGE)
+            .hasNumberOfSubReports(0);
+
+        assertThat(configurationReport.getName())
+            .isEqualTo(ReporterCoreKey.GENERAL_TEST_CLASS_CONFIGURATION_REPORT);
+
+        assertThat(testMethodReports).size().isEqualTo(2);
+    }
+
+    @Test
+    public void verify_test_class_configuration_report_exists_with_contents() {
+        List<TestClassReport> testClassReports = executionReport.getTestSuiteReports().get(0).getTestClassReports();
+        TestClassReport testClassReport = testClassReports.get(0);
+        ConfigurationReport configurationReport = testClassReport.getConfiguration();
+        Report deploymentReport = configurationReport.getSubReports().get(0);
+
+        assertThatReport(configurationReport)
+            .hasName(ReporterCoreKey.GENERAL_TEST_CLASS_CONFIGURATION_REPORT)
+            .hasNumberOfEntries(0)
+            .hasNumberOfSubReports(1);
+
+        verify_deployment_configuration_sub_report(deploymentReport);
+    }
+
+    private void verify_deployment_configuration_sub_report(Report deploymentReport) {
+        Report nestedDeploymentReport = (Report) deploymentReport.getSubReports().get(0);
+
+        assertThat(deploymentReport.getName())
+            .isEqualTo(new UnknownStringKey("deployments"));
+        assertThat(deploymentReport.getEntries()).size().isEqualTo(0);
+        assertThat(deploymentReport.getSubReports()).size().isEqualTo(1);
+
+        assertThatReport(nestedDeploymentReport)
+            .hasName(ArquillianCoreKey.DEPLOYMENT_IN_TEST_CLASS_REPORT)
+            .hasNumberOfEntries(4)
+            .hasEntryContainingKeys(ArquillianCoreKey.DEPLOYMENT_IN_TEST_CLASS_NAME,
+                ArquillianCoreKey.ARCHIVE_NAME_OF_DEPLOYMENT,
+                ArquillianCoreKey.ORDER_OF_DEPLOYMENT,
+                ArquillianCoreKey.PROTOCOL_USED_FOR_DEPLOYMENT)
```

Comment:
name, order and protocol of the deployment won't change on other platforms, which means that you can verify also the values. If you set also the archive name here: https://github.com/arquillian/arquillian-reporter/pull/17/files#diff-5316e8921f2cf4c1d52e2915139433d6R20 then you can verify also that one